File: arm.64/langs/php/src/bind.php
<?php
function bind(string $binaryPath): ?object {
    $filename = basename($binaryPath);
    $skips = ['c', 'cpp', 'rust', 'go', 'swift', 'haskell', 'zig', 'v', 'slint_sample'];
    if (in_array($filename, $skips)) return null;

    try {
        $ffi = FFI::cdef("
            typedef const char* Address;
            typedef const char* Payload;
            typedef const char* Options;
            int Pump(Address address, Payload payload, Options options);
        ", $binaryPath);

        return (object)[
            'name' => $filename,
            'path' => $binaryPath,
            'ffi'  => $ffi
        ];
    } catch (Throwable $e) {
        return null;
    }
}

File: arm.64/langs/lua/src/bind.lua
local ffi = require("ffi")

local M = {}

function M.bind(binary_path)
    local filename = binary_path:match("([^/]+)$")
    local skips = { c=true, cpp=true, rust=true, go=true, swift=true, haskell=true, zig=true, v=true, slint_sample=true }
    if skips[filename] then return nil end

    local ok, lib = pcall(ffi.load, binary_path)
    if not ok then return nil end

    local pump_ok, pump = pcall(function() return lib.Pump end)
    if not pump_ok or not pump then return nil end

    return {
        name = filename,
        path = binary_path,
        lib = lib,
        Pump = pump
    }
end

return M
